packages/forms/src/directives/validators.ts

```diff
@@ -652,44 +652,22 @@ export const PATTERN_VALIDATOR: any = {
 @Directive({
   selector: '[pattern][formControlName],[pattern][formControl],[pattern][ngModel]',
   providers: [PATTERN_VALIDATOR],
-  host: {'[attr.pattern]': 'pattern ? pattern : null'}
+  host: {'[attr.pattern]': '_enabled ? pattern : null'}
 })
-export class PatternValidator implements Validator, OnChanges {
-  private _validator: ValidatorFn = nullValidator;
-  private _onChange?: () => void;
-
+export class PatternValidator extends AbstractValidatorDirective {
   /**
    * @description
    * Tracks changes to the pattern bound to this directive.
    */
   @Input()
   pattern!: string|RegExp;  // This input is always defined, since the name matches selector.
 
-  /** @nodoc */
-  ngOnChanges(changes: SimpleChanges): void {
-    if ('pattern' in changes) {
-      this._createValidator();
-      if (this._onChange) this._onChange();
-    }
-  }
-
-  /**
-   * Method that validates whether the value matches the pattern requirement.
-   * @nodoc
-   */
-  validate(control: AbstractControl): ValidationErrors|null {
-    return this._validator(control);
-  }
+  /** @internal */
+  override inputName = 'pattern';
 
-  /**
-   * Registers a callback function to call when the validator inputs change.
-   * @nodoc
-   */
-  registerOnValidatorChange(fn: () => void): void {
-    this._onChange = fn;
-  }
+  /** @internal */
+  override normalizeInput = (input: string|RegExp): ValidatorFn => patternValidator(input);
```

Comment:
In case of the PatternValidator, there is no need to normalize an input, so the function could look like this:
```suggestion
  override normalizeInput = (input: string|RegExp): string|RegExp => input;
```